### PR TITLE
Update command in section Configuring Capsule Server to Generate and …

### PR DESCRIPTION
**Pull request by suyogsainkar**
…Sign Puppet Certificates

-since Sat6.8 uses Puppet 6

Puppet command needs to be updated as 6.8 uses puppet6
https://bugzilla.redhat.com/show_bug.cgi?id=1865905

### CODE REVIEW

**Comment by akhil-jha** [comment]
Looks good to me.

File: guides/doc-Configuring_Load_Balancer/topics/Configuring_Capsule_Server_with_Default_SSL_Certificates_for_Load_Balancing_with_Puppet.adoc [hunk 1]
```diff
@@ -82,6 +90,13 @@ This command creates the following files on the system where you configure {Smar
 * `/etc/puppetlabs/puppet/ssl/certs/_{smartproxy-example-com}_.pem`
 * `/etc/puppetlabs/puppet/ssl/private_keys/_{smartproxy-example-com}_.pem`
 * `/etc/puppetlabs/puppet/ssl/public_keys/_{smartproxy-example-com}_.pem`
+.. Resume the Puppet agent:
```
Comment (ekohl) [inline guides/doc-Configuring_Load_Balancer/topics/Configuring_Capsule_Server_with_Default_SSL_Certificates_for_Load_Balancing_with_Puppet.adoc:93]: You're starting the server, not the agent.

File: guides/doc-Configuring_Load_Balancer/topics/Configuring_Capsule_Server_with_Default_SSL_Certificates_for_Load_Balancing_with_Puppet.adoc [hunk 2]
```diff
@@ -68,12 +68,20 @@ root@_{smart-proxy-context}_ca.example.com_:__{smart-proxy-context}_ca.example.c
...
-# puppet cert generate _{smartproxy-example-com}_ \
---dns_alt_names=_loadbalancer.example.com_
+# puppet resource service puppetserver ensure=stopped
+----
++
+.. Generate certificate for the {SmartProxyServer}:
++
+[options="nowrap", subs="+quotes,verbatim,attributes"]
+----
+# puppetserver ca generate --certname _{smartproxy-example-com}_ \
```
Comment (ekohl) [inline guides/doc-Configuring_Load_Balancer/topics/Configuring_Capsule_Server_with_Default_SSL_Certificates_for_Load_Balancing_with_Puppet.adoc:83]: Why is this needed? You're already passing all the options in the installer command. This doesn't make sense to me unless there's a bug.

**Review by ekohl — CHANGES_REQUESTED** [review]
If this is needed, it's a bug. Please don't document workarounds.

File: guides/doc-Configuring_Load_Balancer/topics/Configuring_Capsule_Server_with_Default_SSL_Certificates_for_Load_Balancing_with_Puppet.adoc (hunk 2 above)
Comment (ekohl) [reply guides/doc-Configuring_Load_Balancer/topics/Configuring_Capsule_Server_with_Default_SSL_Certificates_for_Load_Balancing_with_Puppet.adoc:83]: Apologies, I was a bit too quick. Looks like this may indeed be needed. I'm not sure if the other Puppetservers are configured to use the bits that use `--ca-client`.

I would also add a note that `--ca-client` should only be used for the load balanced servers. If people use it to create certificates for regular nodes, it creates a big security issue.

File: guides/doc-Configuring_Load_Balancer/topics/Configuring_Capsule_Server_with_Default_SSL_Certificates_for_Load_Balancing_with_Puppet.adoc [hunk 3]
```diff
@@ -68,12 +68,20 @@ root@_{smart-proxy-context}_ca.example.com_:__{smart-proxy-context}_ca.example.c
 --enable-foreman-proxy-plugin-remote-execution-ssh
 ----
 
-. On {SmartProxyServer}, generate Puppet certificates for all other {SmartProxyServer}s that you configure for load balancing, except this first system where you configure Puppet certificates signing:
+. On {SmartProxyServer}, generate Puppet certificates for all other {SmartProxyServer}s that you configure for load balancing, except the first system where you configure Puppet certificates signing:
+.. First, stop the Puppet agent:
```
Comment (ekohl) [inline guides/doc-Configuring_Load_Balancer/topics/Configuring_Capsule_Server_with_Default_SSL_Certificates_for_Load_Balancing_with_Puppet.adoc:72]:
```suggestion
.. First, stop the Puppetserver:
```

**Comment by akhil-jha** [comment]
> Looks good to me.

I am sorry, I missed the `puppet agent -> puppetserver` part.
@ekohl ,thanks for adding. Yes, `--ca-client` to be only used with loadbalancers.

File: guides/doc-Configuring_Load_Balancer/topics/Configuring_Capsule_Server_with_Default_SSL_Certificates_for_Load_Balancing_with_Puppet.adoc (hunk 2 above)
Comment (suyogsainkar) [reply guides/doc-Configuring_Load_Balancer/topics/Configuring_Capsule_Server_with_Default_SSL_Certificates_for_Load_Balancing_with_Puppet.adoc:83]: No worries. Added a warning admonition/note. thanks.

File: guides/doc-Configuring_Load_Balancer/topics/Configuring_Capsule_Server_with_Default_SSL_Certificates_for_Load_Balancing_with_Puppet.adoc (hunk 1 above)
Comment (suyogsainkar) [reply guides/doc-Configuring_Load_Balancer/topics/Configuring_Capsule_Server_with_Default_SSL_Certificates_for_Load_Balancing_with_Puppet.adoc]: Fixed

File: guides/doc-Configuring_Load_Balancer/topics/Configuring_Capsule_Server_with_Default_SSL_Certificates_for_Load_Balancing_with_Puppet.adoc (hunk 3 above)
Comment (suyogsainkar) [reply guides/doc-Configuring_Load_Balancer/topics/Configuring_Capsule_Server_with_Default_SSL_Certificates_for_Load_Balancing_with_Puppet.adoc]: Fixed

**Comment by suyogsainkar** [comment]
@akhil-jha @ekohl thanks for your review and comments. I've applied your suggestions.

**Comment by suyogsainkar** [comment]
@spetrosi can you please peer review the updates. Thanks.

File: guides/doc-Configuring_Load_Balancer/topics/Configuring_Capsule_Server_with_Default_SSL_Certificates_for_Load_Balancing_with_Puppet.adoc [hunk 4]
```diff
@@ -68,21 +68,41 @@ root@_{smart-proxy-context}_ca.example.com_:__{smart-proxy-context}_ca.example.c
...
 +
+.. Generate certificate for the {SmartProxyServer}:
++
+[options="nowrap", subs="+quotes,verbatim,attributes"]
+----
+# puppetserver ca generate --certname _{smartproxy-example-com}_ \
+--subject-alt-names _loadbalancer.example.com_ --ca-client
+----
++
+WARNING: The '--ca-client' option in the command must only be used for server load balancers. Using the '--ca-client' option to create certificates for regular nodes can create security issues.
```
Comment (spetrosi) [inline guides/doc-Configuring_Load_Balancer/topics/Configuring_Capsule_Server_with_Default_SSL_Certificates_for_Load_Balancing_with_Puppet.adoc:87]: The warning is not needed because it is stated that you must complete the procedure on the Puppet master only in the introduction: `Complete this procedure only for the system where you want to configure Smart Proxy server to generate and sign Puppet certificates for all other Smart Proxy servers that you configure for load balancing.`

File: guides/doc-Configuring_Load_Balancer/topics/Configuring_Capsule_Server_with_Default_SSL_Certificates_for_Load_Balancing_with_Puppet.adoc [hunk 5]
```diff
@@ -68,21 +68,41 @@ root@_{smart-proxy-context}_ca.example.com_:__{smart-proxy-context}_ca.example.c
...
+.. Resume the Puppet server:
++
+[options="nowrap", subs="+quotes,verbatim,attributes"]
+----
+# puppet resource service puppetserver ensure=running
+----
++
+
+
+
```
Comment (spetrosi) [inline guides/doc-Configuring_Load_Balancer/topics/Configuring_Capsule_Server_with_Default_SSL_Certificates_for_Load_Balancing_with_Puppet.adoc:105]: a redundant `+` and extra new lines here

File: guides/doc-Configuring_Load_Balancer/topics/Configuring_Capsule_Server_with_Default_SSL_Certificates_for_Load_Balancing_with_Puppet.adoc [hunk 6]
```diff
@@ -68,21 +68,41 @@ root@_{smart-proxy-context}_ca.example.com_:__{smart-proxy-context}_ca.example.c
 --enable-foreman-proxy-plugin-remote-execution-ssh
 ----
 
-. On {SmartProxyServer}, generate Puppet certificates for all other {SmartProxyServer}s that you configure for load balancing, except this first system where you configure Puppet certificates signing:
+. On {SmartProxyServer}, generate Puppet certificates for all other {SmartProxyServer}s that you configure for load balancing, except the first system where you configure Puppet certificates signing:
+.. First, stop the Puppet server:
```
Comment (spetrosi) [inline guides/doc-Configuring_Load_Balancer/topics/Configuring_Capsule_Server_with_Default_SSL_Certificates_for_Load_Balancing_with_Puppet.adoc:72]:
```suggestion
.. Stop the Puppet server:
```
It is the first substep, `first` adds no meaning here.

File: guides/doc-Configuring_Load_Balancer/topics/Configuring_Capsule_Server_with_Default_SSL_Certificates_for_Load_Balancing_with_Puppet.adoc [hunk 7]
```diff
@@ -68,21 +68,41 @@ root@_{smart-proxy-context}_ca.example.com_:__{smart-proxy-context}_ca.example.c
...
 +
 [options="nowrap", subs="+quotes,verbatim,attributes"]
 ----
-# puppet cert generate _{smartproxy-example-com}_ \
---dns_alt_names=_loadbalancer.example.com_
+# puppet resource service puppetserver ensure=stopped
 ----
 +
+.. Generate certificate for the {SmartProxyServer}:
```
Comment (spetrosi) [inline guides/doc-Configuring_Load_Balancer/topics/Configuring_Capsule_Server_with_Default_SSL_Certificates_for_Load_Balancing_with_Puppet.adoc:79]: This step should be more descriptive. I think this can be fixed by getting rid of substeps - using `.` for steps to keep them in the procedure and not create an additional indentation. With that, we can make this step extra clear and use the full sentence ` . On {SmartProxyServer}, generate Puppet certificates for all other {SmartProxyServer}s that you configure for load balancing, except the first system where you configure Puppet certificates signing:`. What do you think?

File: guides/doc-Configuring_Load_Balancer/topics/Configuring_Capsule_Server_with_Default_SSL_Certificates_for_Load_Balancing_with_Puppet.adoc (hunk 5 above)
Comment (suyogsainkar) [reply guides/doc-Configuring_Load_Balancer/topics/Configuring_Capsule_Server_with_Default_SSL_Certificates_for_Load_Balancing_with_Puppet.adoc:101]: Fixed

File: guides/doc-Configuring_Load_Balancer/topics/Configuring_Capsule_Server_with_Default_SSL_Certificates_for_Load_Balancing_with_Puppet.adoc (hunk 7 above)
Comment (suyogsainkar) [reply guides/doc-Configuring_Load_Balancer/topics/Configuring_Capsule_Server_with_Default_SSL_Certificates_for_Load_Balancing_with_Puppet.adoc]: Updated, thanks.

**Comment by suyogsainkar** [comment]
@spetrosi I've applied peer review suggestions; ready for another review again. thanks.

File: guides/doc-Configuring_Load_Balancer/topics/Configuring_Capsule_Server_with_Default_SSL_Certificates_for_Load_Balancing_with_Puppet.adoc [hunk 8]
```diff
@@ -68,12 +68,19 @@ root@_{smart-proxy-context}_ca.example.com_:__{smart-proxy-context}_ca.example.c
 --enable-foreman-proxy-plugin-remote-execution-ssh
 ----
 
-. On {SmartProxyServer}, generate Puppet certificates for all other {SmartProxyServer}s that you configure for load balancing, except this first system where you configure Puppet certificates signing:
+. Stop the Puppet server:
```
Comment (spetrosi) [inline guides/doc-Configuring_Load_Balancer/topics/Configuring_Capsule_Server_with_Default_SSL_Certificates_for_Load_Balancing_with_Puppet.adoc:71]:
```suggestion
. On {SmartProxyServer}, stop the Puppet server:
```

File: guides/doc-Configuring_Load_Balancer/topics/Configuring_Capsule_Server_with_Default_SSL_Certificates_for_Load_Balancing_with_Puppet.adoc [hunk 9]
```diff
@@ -68,12 +68,19 @@ root@_{smart-proxy-context}_ca.example.com_:__{smart-proxy-context}_ca.example.c
...
 +
 [options="nowrap", subs="+quotes,verbatim,attributes"]
 ----
-# puppet cert generate _{smartproxy-example-com}_ \
---dns_alt_names=_loadbalancer.example.com_
+# puppet resource service puppetserver ensure=stopped
+----
++
+. On {SmartProxyServer}, generate Puppet certificates for all other {SmartProxyServer}s that you configure for load balancing, except the first system where you configure Puppet certificates signing:
```
Comment (spetrosi) [inline guides/doc-Configuring_Load_Balancer/topics/Configuring_Capsule_Server_with_Default_SSL_Certificates_for_Load_Balancing_with_Puppet.adoc:78]:
```suggestion
. Generate Puppet certificates for all other {SmartProxyServer}s that you configure for load balancing, except the first system where you configure Puppet certificates signing:
```

File: guides/doc-Configuring_Load_Balancer/topics/Configuring_Capsule_Server_with_Default_SSL_Certificates_for_Load_Balancing_with_Puppet.adoc (hunk 8 above)
Comment (suyogsainkar) [reply guides/doc-Configuring_Load_Balancer/topics/Configuring_Capsule_Server_with_Default_SSL_Certificates_for_Load_Balancing_with_Puppet.adoc]: Updated

File: guides/doc-Configuring_Load_Balancer/topics/Configuring_Capsule_Server_with_Default_SSL_Certificates_for_Load_Balancing_with_Puppet.adoc (hunk 9 above)
Comment (suyogsainkar) [reply guides/doc-Configuring_Load_Balancer/topics/Configuring_Capsule_Server_with_Default_SSL_Certificates_for_Load_Balancing_with_Puppet.adoc]: Updated
